common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/responses/ApiErrorResult.kt

```diff
@@ -0,0 +1,37 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.providers.nativeauth.responses
+
+import com.microsoft.identity.common.java.logging.DiagnosticContext
+
+
+/**
+ * Base class to encapsulate various error response from the Native Auth API
+ */
+open class ApiErrorResult(
+    open val error: String?,
+    open val errorDescription: String?,
+    open val details: List<Map<String, String>>? = null,
+    open val correlationId: String = DiagnosticContext.INSTANCE.threadCorrelationId,
```

Comment:
eSTS should be sending you a correlation id back. You should trust that one more than the one on the client

Reply:
Hey @shahzaibj , do we currently send the client correlation ID as part of our request to ESTS? If this is done, does ESTS use the ID we send, rather than generating a new one? Or does ESTS server-side always generate a correlation ID on it's end?